docs/schema-changes.md

```diff
@@ -0,0 +1,119 @@
+registry, as well as the diffing process. This diff report can be used to:
+
+- Understand the changes between two versions of a semantic convention registry.
+- Update the OpenTelemetry Schema file, section versions, with the new version.
+- Generate a migration guide for users of the semantic convention registry.
+- Generate a SQL DDL script to update a database schema.
+- And more.
+
+> Note: The current implementation of the diffing process focuses on the top-level
+items attributes, metrics, events, spans, resources and does not compare the fields
+of those top-level items.
+
+## Data Model
+
+The schema changes data model is composed of the following components:
+
+- `head`: The registry manifest of the most recent version of the semantic
+  convention registry used in the diffing process.
+- `baseline`: The registry manifest of the oldest version of the semantic
+  convention registry used in the diffing process.
+- `changes`: A dictionary of changes between the head and baseline registries.
+  The dictionary is composed of the following keys (when applicable):
+  - `attributes`: A list of changes to attributes.
+  - `metrics`: A list of changes to metrics.
+  - `events`: A list of changes to events.
+  - `spans`: A list of changes to spans.
+  - `resources`: A list of changes to resources.
+
+Each change in the changes dictionary for any key is represented as a list of
+schema changes, represented by one of the following types:
+
+- `added`: A new schema item (e.g., attribute, metric, etc.) was added in the
+  head registry. The name of the new item is stored in the name attribute.
+- `renamed_to_new`: One or more schema items in the baseline registry were
+  renamed to the same new name in the head registry. The old names of the
+  items are stored in the old_names attribute, and the new name is stored in
+  the current_name attribute.
+- `renamed_to_existing`: One or more schema items in the baseline registry were
+  renamed to an existing item in the head registry. The old names of the items
+  are stored in the old_names attribute, and the existing item name is stored
+  in the current_name attribute.
+- `deprecated`: An item in the baseline registry was deprecated in the head
+  registry. The name of the deprecated item is stored in the name attribute,
+  and the deprecation note is stored in the note attribute.
+- `removed`: An item in the baseline registry was removed in the head
+  registry. The name of the removed item is stored in the name attribute.
+
+> Note: Although the removed schema change type is a valid output of the diffing
+process, it should never be present in the diff report between two versions of
+well-formed semantic convention registries. The policy for semantic convention
+registries is to deprecate items instead of removing them.
+
+Example Schema Diff in YAML
+
+```yaml
+head:
+  semconv_version: v1.27.0
+baseline:
+  semconv_version: v1.26.0
+changes:
+  attributes:
+    - type: deprecated
```

Comment:
nit and personal taste, so definitely not blocking

having 

```
- name: http.server_name
  type: deprecated
```

looks more readable to me, maybe because it's closer to how we write attribute definitions.

Reply:
Ok I will make this change in the documentation.

Reply:
Updated the documentation